WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -1629,17 +1501,68 @@ local methods = {
           if not (self.data.skipWagoUpdate and self.data.skipWagoUpdate == updateData.wagoVersion) then
             if tonumber(updateData.wagoVersion) > tonumber(self.data.version) then
               -- got update
-              return true, false, updateData
+              return true, false, updateData, slug
             end
           else
             -- version skip flag
-            return true, true, updateData
+            return true, true, updateData, slug
           end
         end
       end
     end
     -- no addon, or no data, or ignore flag
-    return false, false, nil
+    return false, false, nil, nil
+  end,
+  ["RefreshUpdate"] = function(self, actionFunc)
+    if self.data.parent then
+      -- is in a group
+      local parentButton = WeakAuras.GetDisplayButton(self.data.parent)
+      if parentButton then
+        parentButton:RefreshUpdate(actionFunc)
+      end
+    else
+      if self.data.controlledChildren then
+        -- is a group
+        local hasUpdate, skipVersion, _, slug = self:HasUpdate()
+        local showGroupUpdateIcon = false
+        for childIndex, childId in pairs(self.data.controlledChildren) do
+          local childButton = WeakAuras.GetDisplayButton(childId);
+          if childButton then
+            if actionFunc then
+              childButton.callbacks[actionFunc](nil, true)
+            end
+            childButton:RefreshUpdateMenu()
+            local childHasUpdate, childSkipVersion, _, childSlug = childButton:HasUpdate()
+            if childHasUpdate and slug ~= childSlug and not childSkipVersion then
+              showGroupUpdateIcon = true
+              childButton:ShowUpdateIcon()
+            else
+              childButton:HideUpdateIcon()
+            end
+          end
+        end
+        self:RefreshUpdateMenu()
```

Comment:
These lines here are identically between groups / top level.

For both you are calling RefreshUpdateMenu + Show/HideUpdateIcon, and for both your arecalling self:HasUpdate.

So what you can do is move the if self.data.controlledChildren down to just after the line "local showGroupUpdateIcon = false". And move the end of that if to just above self:RefreshMenu.